AMM/project/scripts/swap-exact-input/src/main.sw

```diff
@@ -0,0 +1,58 @@
+script;
+
+use libraries::{AMM, Exchange};
+use std::block::height;
+
+enum InputError {
+    RouteTooShort: (),
+    SwapAmountsNotExact: (),
+}
+
+enum SwapError {
+    PairExchangeNotRegistered: (ContractId, ContractId),
+}
+
+fn main(
+    amm_contract_address: ContractId,
+    assets: Vec<ContractId>,
+    amounts: Vec<u64>,
```

Comment:
IMO it would be cleaner to take in a single `Vec<Amounts>` where `Amounts` (or whatever you want to name it) is defined as

```
struct Amounts {
    id: ContractId,
    amount: u64,
}
```

Reply:
Removed `amounts` altogether.